### ISSUE TITLE
[arcface_torch]keep getting fp16 gradscale is 0 when training with partial fc

### ISSUE THREAD

**Issue by twmht** [deepinsight/insightface#1437]
Hi,

I tried to train glint360k with arcface-torch.

but the gradscale keeps getting zero and the loss keeps unchanged in each iteration.

any idea?

**Comment by anxiangsir** [comment]
Can you reimplement training for default config?
Your torch version?

**Comment by twmht** [comment]
my torch version is 1.6.

the network is resnet50. and the training dataset is glint360K.

**Comment by twmht** [comment]
when training emore, the loss still keeps 69.0776. it's same with glint360k.

**Comment by twmht** [comment]
change to pytorch1.7. and change fp16 to False. the loss still keeps getting 69.0776.

**Comment by twmht** [comment]
ok.

I remove Dali (https://docs.nvidia.com/deeplearning/dali/user-guide/docs/) and use MXFaceDataSet, the problem is gone.

what is the magic behind MXFaceDataSet?

with Dali I can speed up the model training by 2~4 times.

**Comment by twmht** [comment]
solved. the root cause is that I don't convert label to long type

**Comment by tianxianhao** [comment]
nice! you solved my problem. 
thanks!
